javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithExtends.java

```diff
@@ -0,0 +1,37 @@
+package com.github.javaparser.ast.nodeTypes;
+
+import java.util.List;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.type.ClassOrInterfaceType;
+
+public interface NodeWithExtends<T> {
+    public List<ClassOrInterfaceType> getExtends();
+
+    public T setExtends(final List<ClassOrInterfaceType> extendsList);
+
+    /**
+     * Add an extends to this and automatically add the import
+     * 
+     * @param clazz the class to extand from
+     * @return this
+     */
+    public default T addExtends(Class<?> clazz) {
+        ((Node) this).tryAddImportToParentCompilationUnit(clazz);
+        return addExtends(clazz.getSimpleName());

```

Comment:
Hello, to the idea to provide new Node Type, why to not pass Node as an Interface ?
I certainly too young on this project to see the side effect of this refractor.
In that way, you can do this, 
`public interface NodeWithExtends<T> implements Node {`

I say because, if I clreate a new class like that 
```
public class Foo implements NodeWithExtends<EnumDeclaration> {
  public List<ClassOrInterfaceType> getExtends(){/*implement*/}
public T setExtends(final List<ClassOrInterfaceType> extendsList){/*implement*/}
}
```

Your default method throw a ClassCastException